cmds/tr/tr.go

```diff
@@ -0,0 +1,306 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// tr - translate or delete characters
+
+// Synopsis:
+//     tr [OPTION]... SET1 [SET2]
+
+// Description:
+//     Translate, squeeze, and/or delete characters from standard input, writing
+//     to standard output.
+//
+//     -d: delete characters in SET1, do not translate
+//
+// SETs  are  specified  as  strings of characters. Most represent themselves.
+// Interpreted sequences are:
+//     \\        backslash
+//     \a        audible BEL
+//     \b        backspace
+//     \f        form feed
+//     \n        new line
+//     \r        return
+//     \t        horizontal tab
+//     \v        vertical tab
+//     [:alnum:] all letters and digits
+//     [:alpha:] all letters
+//     [:digit:] all digits
+//     [:graph:] all printable characters
+//     [:cntrl:] all control characters
+//     [:lower:] all lower case letters
+//     [:upper:] all upper case letters
+//     [:space:] all whitespaces
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"unicode"
+)
+
+var delete = flag.Bool("d", false, "delete characters in SET1, do not translate")
+
+const name = "tr"
+
+var escapeChars = map[rune]rune{
+	'\\': '\\',
+	'a':  '\a',
+	'b':  '\b',
+	'f':  '\f',
+	'n':  '\n',
+	'r':  '\r',
+	't':  '\t',
+	'v':  '\v',
+}
+
+type Set int
+
+const (
+	ALNUM Set = iota
```

Comment:
just make these constants the strings themselves, I think, then you don't even need cmdToSet below and no other code anywhere will change. 

You can name the type if you want
type Set string

const (
	ALNUM Set = "[:alnum:]"

etc.
)

you can use this in case, compares, and so on. It's not a high performance need so you don't need to worry about slowness.

Reply:
nice idea. Went further and created the map. That allowed make setToRune method trivial